src/script/lua_api/l_mapgen.cpp

```diff
@@ -527,30 +525,12 @@ int ModApiMapgen::l_get_heat(lua_State *L)
 
 	v3s16 pos = read_v3s16(L, 1);
 
-	NoiseParams np_heat;
-	NoiseParams np_heat_blend;
-
-	MapSettingsManager *settingsmgr =
-		getServer(L)->getEmergeManager()->map_settings_mgr;
+	const BiomeGen *biomegen = getServer(L)->getEmergeManager()->getBiomeGen();
```

Comment:
What if this function is called on script init? Segfault due to missing emerge mananger?